Updates for ece-tools error reference (#7578)
magento/devdocs: src/cloud/reference/ece-tools-error-reference.md

In the `cloud.error.log` file, each entry is formatted as a JSON string for easier parsing:

```json
{"errorCode":1006,"stage":"build","step":"validate-config","suggestion":"No stores/website/locales found in config.php\n  To speed up the deploy process do the following:\n  1. Using SSH, log in to your Magento Cloud account\n  2. Run \"php ./vendor/bin/ece-tools config:dump\"\n  3. Using SCP, copy the app/etc/config.php file to your local repository\n  4. Add, commit, and push your changes to the app/etc/config.php file","title":"The configured state is not ideal","type":"warning"}
```

Error messages are categorized by deployment stage–*build*, *deploy*, and *post-deploy*. Each section provides a list of associated errors with the following information for each error:

-  **Error code**:  The Magento-assigned identifier for the error message
-  **Stage**:  Indicates whether the error occurred during the build, deploy, or post-deploy stage

> -  **Step**:  Indicates the step in the deployment scenario where the error can occur. If the _Step_ column is blank, the error is a general error that can occur in multiple steps, or during pre-processing operations. See [Scenario-based deployment]({{ site.baseurl }}/cloud/deploy/scenario-based-deployment.html) for more information about the build, deploy, and post-deploy steps.

Author: ```suggestion
-  **Step**:  Indicates the step in the deployment scenario that can return the error. If the _Step_ column is blank, the error is a general error that can be returned in multiple steps, or during pre-processing operations. See [Scenario-based deployment]({{ site.baseurl }}/cloud/deploy/scenario-based-deployment.html) for more information about the build, deploy, and post-deploy steps.
```
